activerecord/test/cases/adapters/postgresql/uuid_test.rb

```diff
@@ -403,3 +403,41 @@ def test_find_by_with_uuid
     assert_nil UuidPost.find_by(id: 789)
   end
 end
+
+class PostgresqlUUIDCaseInsensitive < ActiveRecord::PostgreSQLTestCase
+  include PostgresqlUUIDHelper
+
+  class UuidModel < ActiveRecord::Base
+    self.table_name = "pg_uuid_models"
+  end
+
+  setup do
+    connection.transaction do
+      connection.create_table("pg_uuid_models", id: :uuid, **uuid_default) do |t|
+        t.string "title"
+      end
+    end
+  end
+
+  teardown do
+    drop_table "pg_uuid_models"
+  end
+
+  def test_upcase_not_dirty
+    model = UuidModel.create!(id: "abcd-0123-4567-89ef-dead-beef-0101-1010")
+    model.id = "ABCD-0123-4567-89EF-DEAD-BEEF-0101-1010"
+    assert_not model.changed?
+  end
+
+  def test_different_uuid_is_dirty
+    model = UuidModel.create!(id: "abcd-0123-4567-89ef-dead-beef-0101-1010")
+    model.id = "0000-1111-ABCD-EF01-9999-8888-2323-4545"
+    assert model.changed?
```

Comment:
I'm not sure it's necessary to test this here.  This *should be* already covered by other Active Record tests.

Reply:
Sounds good